apis/appmesh/v1beta2/virtualrouter_types.go

```diff
@@ -41,73 +41,32 @@ type WeightedTarget struct {
 	Weight int64 `json:"weight"`
 }
 
-type MatchRange struct {
-	// The start of the range.
-	// +optional
-	Start int64 `json:"start"`
-	// The end of the range.
-	// +optional
-	End int64 `json:"end"`
-}
-
-// HeaderMatchMethod refers to https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HeaderMatchMethod.html
-type HeaderMatchMethod struct {
-	// The value sent by the client must match the specified value exactly.
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=255
-	// +optional
-	Exact *string `json:"exact,omitempty"`
-	// The value sent by the client must begin with the specified characters.
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=255
-	// +optional
-	Prefix *string `json:"prefix,omitempty"`
-	// An object that represents the range of values to match on.
-	// +optional
-	Range *MatchRange `json:"range,omitempty"`
-	// The value sent by the client must include the specified characters.
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=255
-	// +optional
-	Regex *string `json:"regex,omitempty"`
-	// The value sent by the client must end with the specified characters.
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=255
-	// +optional
-	Suffix *string `json:"suffix,omitempty"`
-}
-
-// HTTPRouteHeader refers to https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HttpRouteHeader.html
-type HTTPRouteHeader struct {
-	// A name for the HTTP header in the client request that will be matched on.
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=50
-	Name string `json:"name"`
-	// The HeaderMatchMethod object.
-	// +optional
-	Match *HeaderMatchMethod `json:"match,omitempty"`
-	// Specify True to match anything except the match criteria. The default value is False.
-	// +optional
-	Invert *bool `json:"invert,omitempty"`
-}
-
 // HTTPRouteMatch refers to https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HttpRouteMatch.html
 type HTTPRouteMatch struct {
 	// An object that represents the client request headers to match on.
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=10
 	// +optional
-	Headers []HTTPRouteHeader `json:"headers,omitempty"`
+	Headers []HTTPGatewayRouteHeader `json:"headers,omitempty"`
```

Comment:
Why the rename if the struct is used by both VR and VG? Our naming pattern should match with AppMesh..

Reply:
appmesh sdk has two different structs even though they have same functionality. HttpRouteHeader andHttpGatewayRouteHeader even though they are identical. On the controller side, I tried tohave only 1 copy common to both